src/stransi/escape.py

```diff
@@ -39,7 +39,7 @@ def tokens(self) -> Iterator[Token]:
             if not param:
                 yield Token(kind=kind, data=0)
                 continue
-            yield Token(kind=kind, data=int(param))
+            yield Token(kind=kind, data=int(param.replace('?', '-')))
```

Comment:
I don't like that. It would be better if we had a boolean flag in `Token` https://github.com/getcuia/stransi/blob/c4a395fbe9144574cc8fd48b3eb99a502de47395/src/stransi/token.py#L9-L40

Reply:
`is_private` with a false default value, true for private codes. What do you think? @kj7rrv

Reply:
The only problem with that is an existing program looking for a non-private code would start detecting the private code with the same number.

Reply:
Why is that? Couldn't they check the `is_private` flag?

My concern is that we might need to add yet another dimension to `Token` in the future, so mapping to different ranges of integers doesn't scale well.

Reply:
They could, but existing programs don't. Maybe both? Negative means "This isn't a normal non-private code", and then they have to check why?

Reply:
I don't know of any particular ones, but this could affect any program that uses stransi, checks the `data` value to determine what to do with a code ,and sometimes receives private codes as input. I can do this if you want, but it will be an API-breaking change.

Reply:
That's OK, we're under 1.0.0 at the moment.

Reply:
Okay, I'll do it. It looks like not all private codes use a `?` (most do, but a few use other characters). I'll leave the detection code as-is for now, but I'll add a `private_designator` attribute to `Token` that will always be `?` for now (if the token is private; it will be empty for non-private tokens), in addition to `is_private`.

Reply:
Done in 5efbe17